src/memory_pool_internal.h

```diff
@@ -24,6 +26,9 @@ typedef struct umf_memory_pool_t {
 
     // Memory provider used by the pool.
     umf_memory_provider_handle_t provider;
+
+    // Tells whether memory provider is owned by the pool.
+    bool own_provider;
```

Comment:
an alternative approach could be a pointer to the "destructor function" instead of just storing a `bool` flag.
But it is a minor comment and we can keep it as is at least now.